### PR TITLE
Implementation: ViTKD

### PR DESCRIPTION
**Pull request by nda111**
On branch dev-baselines-vitkd
Changes to be committed:
	new file:   mdistiller/distillers/ViTKD.py
	modified:   mdistiller/distillers/__init__.py
	modified:   mdistiller/engine/cfg.py

### CODE REVIEW

File: mdistiller/distillers/ViTKD.py [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+        
+        for layer in self.m_layers:
+            adapter = self.adapters[f'{layer:02d}']
+            generator = self.generators[f'{layer:02d}']
+            a_s = adapter(feature_student['feats'][layer])
+            g_s = generator(a_s)
+            mask = torch.rand_like(g_s[..., 0:1], device=g_s.device) <= self.masking_ratio
+            mask = mask.expand_as(g_s)
+            loss = loss + F.mse_loss(
+                g_s[mask], feature_teacher['feats'][layer][mask],
```
Comment (youngyoonii) [inline mdistiller/distillers/ViTKD.py:100]: ViTKD에서는 Maksed 된 Student를 generator를 거친 후에 Teacher랑 같아지게끔 하는데 이 코드는 adapter generator지난 output에 masking을 씌우는 형태인 거 같아요

스크린샷 2025-06-02 15-30-30

여기서 말하는 내용은 masked token이 learnable token이라고 하고 Loss 계산할 때 이 masked token의 generate  결과만 loss 계산한다고 이해했는데 그거랑 조금 구현이 다른것 같습니다

**Review by youngyoonii — COMMENTED** [review]
Review About ViTKD
